### ISSUE TITLE
Japanese Bravely Second shows garbage as name under regionfree on US n3ds

### ISSUE THREAD

**Issue by chaoskagami** [smealum/3ds_hb_menu#26]
More specifically, the title shows up as "Ōì¤Ōêū»«óÉ" which is obviously not correct. I can't say what it shows up as on a japanese 3ds, unfortunately - but this is obviously a character encoding issue. My best guess is either a) the game is JIS and it's showing as UTF8 or b) formatting title as ascii when it shouldn't.

I notice in the regionfree.c file that no conversion of character set occurs, and the info struct is directly filled with a pointer offset. Are localized game titles handled correctly?

**Comment by yellows8** [comment]
Yeah it's just displaying the strings as ASCII basically atm, don't know what all characters this font includes.

**Comment by HandsomeMatt** [comment]
Try #14 but we probably also need the various characters in the font too.

**Comment by chaoskagami** [comment]
I'll have to set up a cross-toolchain. I have an aversion to devkitpro, since I use a source-based system.

When I finish setting it up, I'll build PR#14 and report differences. Should I be using the master of ctrulib to build this, or a specific commit/tag?

**Comment by HandsomeMatt** [comment]
master should compile fine.

**Comment by chaoskagami** [comment]
Good to know. Once crosstool-ng finishes bootstrapping GCC, I'll report back.

**Comment by profi200** [comment]
It simply does not support special characters yet. Only ASCII. No idea if anyone is working on fixing that.

**Comment by chaoskagami** [comment]
@profi200 No? Maybe I'll take a stab at getting gamecarts' titles working, then. Shouldn't be terribly hard. Just need to figure out what encoding I should convert from to display it.

Also, #14 allows smdh localization properly. It does not however apply cleanly on master and allow building. There's a few references to utils.h which is deleted by #14. Removing said references allows it all to build without issue.

**Comment by chaoskagami** [comment]
Actually, even worse; #14 fucks up titles of games which displayed perfectly fine without the patch. It's a no-go.

**Comment by chaoskagami** [comment]
Incidentally, the font only includes ASCII. No amount of localization will help with that. Furthermore, smea used Segoe UI as the base font. Which is bad. Being a microsoft font and all, it's not okay legally to use it like that, so I replaced it with liberation sans in my branch. I'll submit a PR for that, but it's not nearly as pretty.

Hopefully I'll figure out how to get all the extra glyphs in there soon. :<

**Comment by profi200** [comment]
I just said it's caused by the lack of UTF-8 support and a full font for it.

**Comment by chaoskagami** [comment]
I'm aware. I was just confirming that what you said was correct. Sorry if I came across too forceful or anything. Partially I was just rambling since the codebase isn't well documented, and I was trying to familarize myself with it while half asleep and low on caffiene. Sorry. :(

Anyways, the real problem is the drawCharacter function in text.c. The third parameter is the character to draw, and that's the problem. It could be fixed by replacing with wchar_ts, but that changes a good bit of logic and any strings that aren't wchar_t would have to be rewritten as such. I actually had to deobfuscate the code in text.c file to figure out what it did in the first place, but that's where the issue is.

**Comment by suloku** [comment]
Mashers is developing a font engine with utf8 support, word wrapping, any text color, etc, in fact he is doing it for his homebrew launcher mod.

**Comment by chaoskagami** [comment]
There's a lot of development going on for the 3ds now. It's a lot like the early days of the PSP, which is nice.

Anyways, I'm going to attempt to compile freetype for the 3ds and use that to do the font rendering - I imagine that would make all this work much easier. Guess I'm kinda doing my own thing now, and this bug report has gone way off topic. My fault, of course. ;P

**Comment by profi200** [comment]
There is sftdlib which uses freetype and does text drawing using the GPU.
